### DISCUSSION TITLE
Game Page Flags

### DISCUSSION THREAD

**Discussion by julkasalsa** [RetroAchievements/RAWeb#688] [Ideas]
Every fresh set should be marked with Public Beta flag until it's somewhat verified as stable (could be simply marked/unmarked by developer). WIP flag could be used to mark/unmark sets that are currently under construction (mostly usable to indicate the set is currently revised for all  who are interested in playing this game at the time).

**Comment by kdecks** [comment]
I'd like to expand on this idea and I'd suggested renaming this issue **Game Page Set Status** Two types of set status flags. Both flag will be publicly visible on a game page. This is also related to #102 .
#### 1. Progress Flags:
| Flag|Qualifier for flag|
|---|---|
| ``Empty`` | default|
| ``🔨 Work-in-progress`` | developer has signed in|
| ``Complete`` | developer has finished|
| ``Incomplete`` | developer has left partial|

Clicking on ``🔨 Work-in-progress`` will list all achievements in the *planned* WIP Achievement group, for that set.

The following options in the game page dev tools would be available: 
*(These changes would be audited (#11)*

| Option| Result  |
|---|---|
| Sign in |The progress flag is set automatically from ``Empty`` to ``🔨 Work-in-progress`` by [developer]|
| **(Below all apply if signed in)**|
| Set to Public beta |  **Progress Flag** is set to ``Public Beta``, if there are achievements. |
| Abandon development  |  **Progress Flag** is set back to ``Empty``. If there are no achievements. The developer is signed out. |
| Quit development as feature complete  | **Progress Flag** is set to ``Complete``. The developer is signed out. The set is **locked** from developers making additions. |
| Quit development as partially complete | **Progress Flag** is set to ``Incomplete`` The developer is signed out, the set remains **unlocked** from additions.

#### 2. Stability Flag:
| Flag|Qualifier for flag|
|---|---|
| ``β Public Beta`` | When a **Progress flag** is set to ``Public Beta``|
| ``👶 New set`` |When a **Progress Flag** is set to ``Complete`` or ``Incomplete``. A auto-timer is set for 3 (1-3?) months, after that time the set will convert to ``🌟 Stable``|
|``Has reported problems``| If there are any open tickets younger than one month, or less than 5 open tickets. Would revert to ``👶 New set`` or ``🌟 Stable`` once all tickets are closed |
|``❗️ Unstable``| If there are 5 or more tickets older than one month |
| ``🌟 Stable``| If there are no tickets and the set is not a ``👶 New set``|

Clicking on ``Has reported problems`` and ``❗️ Unstable`` would link directly to the open tickets for the page. This is the ideal place for #7.

(optional) The date of the latest status change would be shown on the game page, possibly in an expandable log form. This log could show when developers have signed into and quit a game.

**Comment by kdecks** [comment]
I've updated this concept to go in tandem with #102 .

**Comment by SyrianBallaS** [comment]
I think this a good idea, this might prevent invalid tickets

**Comment by kdecks** [comment]
An addition: A message that only is show to developers by an `unstable` flag.
```This set has x open ticket you can enhance the community by repairing them.`` 
(or something to this affect.) And link to the open tickets from here.

**Comment by Jamiras** [comment]
Suggest also adding quick reference pages so players can find "public beta" and "new" sets, as well as provide a master list of "in development" games.

**Comment by Jamiras** [comment]
I believe the progress flags are handled by #1027. Leaving open for the possibility of adding a "stability" indicator.
